apps/api/src/app/events/e2e/trigger-event-topic.e2e.ts

```diff
@@ -0,0 +1,261 @@
+import {
+  ChannelTypeEnum,
+  StepTypeEnum,
+  IEmailBlock,
+  TemplateVariableTypeEnum,
+  TopicDto,
+  TopicKey,
+  TopicName,
+  TriggerRecipientsTypeEnum,
+  ExternalSubscriberId,
+} from '@novu/shared';
+import { SubscribersService, UserSession } from '@novu/testing';
+import axios from 'axios';
+import { expect } from 'chai';
+
+import { TriggerEventRequestDto } from '../dtos';
+
+const axiosInstance = axios.create();
+
+const TOPIC_PATH = '/v1/topics';
+const TRIGGER_ENDPOINT = '/v1/events/trigger';
+
+const createTopic = async (session: UserSession, key: TopicKey, name: TopicName): Promise<TopicDto> => {
+  const response = await axiosInstance.post(
+    `${session.serverUrl}${TOPIC_PATH}`,
+    {
+      key,
+      name,
+    },
+    {
+      headers: {
+        authorization: `ApiKey ${session.apiKey}`,
+      },
+    }
+  );
+
+  expect(response.status).to.eql(201);
+  const body = response.data;
+  expect(body.data._id).to.exist;
+
+  return body.data;
+};
+
+const addSubscribersToTopic = async (session: UserSession, topicDto: TopicDto, subscribers: SubscriberEntity[]) => {
+  const subscriberIds: ExternalSubscriberId[] = subscribers.map(
+    (subscriber: SubscriberEntity) => subscriber.subscriberId
+  );
+
+  const response = await axiosInstance.post(
+    `${session.serverUrl}${TOPIC_PATH}/${topicDto._id}/subscribers`,
+    {
+      subscribers: subscriberIds,
+    },
+    {
+      headers: {
+        authorization: `ApiKey ${session.apiKey}`,
+      },
+    }
+  );
+
+  expect(response.status).to.be.eq(204);
+};
+
+const buildTriggerRequestPayload = (
+  template: NotificationTemplateEntity,
+  topicDto: TopicDto,
+  attachments?: Record<string, unknown>[]
+): TriggerEventRequestDto => {
+  const topic: ITopic = { type: TriggerRecipientsTypeEnum.TOPIC, topicId: topicDto._id };
+
+  const payload = {
+    name: template.triggers[0].identifier,
+    to: [topic],
+    payload: {
+      firstName: 'Testing of User Name',
+      urlVariable: '/test/url/path',
+      ...(attachments && { attachments }),
+    },
+  };
+
+  return payload;
+};
+
+const buildTriggerRequestHeaders = (session: UserSession) => ({
+  headers: {
+    authorization: `ApiKey ${session.apiKey}`,
+  },
+});
+
+describe('Trigger event for a topic - /v1/events/trigger (POST)', () => {
+  let session: UserSession;
+  let template: NotificationTemplateEntity;
+  let subscriber: SubscriberEntity;
+  let subscriberService: SubscribersService;
+  let topicDto: TopicDto;
+  let triggerEndpointUrl: string;
+  const notificationRepository = new NotificationRepository();
+  const messageRepository = new MessageRepository();
+  const subscriberRepository = new SubscriberRepository();
+  const integrationRepository = new IntegrationRepository();
+  const logRepository = new LogRepository();
+  const jobRepository = new JobRepository();
+
+  beforeEach(async () => {
+    process.env.FF_IS_TOPIC_NOTIFICATION_ENABLED = 'true';
```

Comment:
Can we enable it by default in all of our existing .env files? .env.test, .env.dev and maybe even in the config make it on by default if not passed otherwise.

Reply:
If we are confident it is not risky and it will get leaked to production I would be ok to do it, but when all the testing is done though.